roiextractors/extractors/caiman/caimansegmentationextractor.py

```diff
@@ -0,0 +1,151 @@
+            if roi_location[0,i]<0:
+                roi_location[0,i]=0
+        return roi_location
+
+    @staticmethod
+    def write_segmentation(segmentation_object, savepath):
+        raise NotImplementedError
+
+    # defining the abstract class enformed methods:
+    def get_roi_ids(self):
+        return list(range(self.no_rois))
+
+    def get_num_rois(self):
+        return self._roi_response.shape[0]
+
+    def get_roi_locations(self, roi_ids=None):
+        if roi_ids is None:
+            return self.roi_locations
+        else:
+            roi_idx = [np.where(np.array(i) == self.roi_ids)[0] for i in roi_ids]
+            ele = [i for i, j in enumerate(roi_idx) if j.size == 0]
+            roi_idx_ = [j[0] for i, j in enumerate(roi_idx) if i not in ele]
+            return self.roi_locations[:, roi_idx_]
+
+    def get_num_frames(self):
+        return self._roi_response.shape[1]
+
+    def get_traces(self, roi_ids=None, start_frame=None, end_frame=None, name='Fluorescence'):
+        if start_frame is None:
+            start_frame = 0
+        if end_frame is None:
+            end_frame = self.get_num_frames() + 1
+        if roi_ids is None:
+            roi_idx_ = range(self.get_num_rois())
+        else:
+            roi_idx = [np.where(np.array(i) == self.roi_ids)[0] for i in roi_ids]
+            ele = [i for i, j in enumerate(roi_idx) if j.size == 0]
+            roi_idx_ = [j[0] for i, j in enumerate(roi_idx) if i not in ele]
+        return np.array([self._roi_response_dict[name][int(i), start_frame:end_frame] for i in roi_idx_])
+
+    def get_roi_image_masks(self, roi_ids=None):
+        if roi_ids is None:
+            roi_idx_ = range(self.get_num_rois())
+        else:
+            roi_idx = [np.where(np.array(i) == self.roi_ids)[0] for i in roi_ids]
+            ele = [i for i, j in enumerate(roi_idx) if j.size == 0]
+            roi_idx_ = [j[0] for i, j in enumerate(roi_idx) if i not in ele]
+        image_mask = np.zeros([np.prod(self.image_size),len(roi_idx_)])
+        for j,i in enumerate(roi_idx_):
+            roi_ids_loop = self._mask_roi_ids[self._mask_ids[i]:self._mask_ids[i+1]]
+            image_mask_loop = self._masks[self._mask_ids[i]:self._mask_ids[i+1]]
+            image_mask[[roi_ids_loop],j] = image_mask_loop
+        return image_mask.reshape(list(self.image_size)+[len(roi_idx_)],order='F')
+
+    def get_roi_pixel_masks(self, roi_ids=None):
+
+        if roi_ids is None:
+            roi_idx_ = self.roi_ids
+        else:
+            roi_idx = [np.where(np.array(i) == self.roi_ids)[0] for i in roi_ids]
+            ele = [i for i, j in enumerate(roi_idx) if j.size == 0]
+            roi_idx_ = [j[0] for i, j in enumerate(roi_idx) if i not in ele]
+        self.pixel_masks = _pixel_mask_extractor(self.get_roi_image_masks(roi_idx_), range(len(roi_idx_)))
+        return self.pixel_masks
+
+    def get_images(self):
+        return {'Images': {'meanImg': self._summary_image_read()}}
```

Comment:
1. `Cn` is the correlation image, not the mean image. It is computed by taking the correlation that each pixel has with each of its neighbors over time. The pixels that have a higher value are generally places where there is a neuron. Cn is useful to get a quick sense of how neurons are dispersed in your image, but doesn't give you segmentation information.
2. As we discussed yesterday, I prefer attributes instead of dicts unless there will be user-defined fields. I would rather have `meanImg` be an attribute of the class.
3. Why the outer `{'Images': ...}`? I understand the inner dict, in case there are other image types (though I would prefer attrs of the class), but I don't see why we have a dict with a single key: `'Images'`.

Reply:
made some changes here: 8d14fecd7902d7d17e390c322dffe768ea4d5b8f
I have unpacked the images and roi_response_dict into independent attributes, defined as None in the base class. 
THey are called in the downstream get methods based on the field that a user desires